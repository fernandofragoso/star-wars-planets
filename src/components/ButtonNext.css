.button, .button--disabled {
  width: 200px;
  outline: 0;
  text-align: center;
  margin-top: 40px;
  padding: 10px 40px;
  font-size: 16px;
  font-weight: bold;
  box-shadow: 5px 5px #000;
  border-radius: 10px;
  background-color: #999;
  border: 0;
  transition: 200ms;
  cursor: pointer;
}

.button:hover {
  background-color: #888;
}

.button:active {
  box-shadow: 0 0;
  transform: translate(5px, 5px);
}

.button--disabled {
  cursor: auto;
  color: #333;
}

@media (max-width: 768px) {
  .button, .button--disabled {
    width: 100%;
    margin-top: 20px;
  }
}
