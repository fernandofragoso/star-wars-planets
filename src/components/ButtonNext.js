import React, { Component } from 'react';
import './ButtonNext.css';

export default class ButtonNext extends Component {
  render() {
    let btClasses = (this.props.loading ? "button--disabled" : "button");

    return (
      <button onClick={this.props.onButtonClick.bind(this)} className={btClasses}>
        NEXT
      </button>
    );
  }
}