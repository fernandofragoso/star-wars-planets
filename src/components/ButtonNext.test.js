import React from 'react';
import ReactDOM from 'react-dom';
import ButtonNext from './ButtonNext';
import renderer from 'react-test-renderer';

let functionMock = jest.fn();

it('renders correctly', () => {
  const loading = false;
  const tree = renderer.create(<ButtonNext loading={loading} onButtonClick={functionMock} />).toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders correctly on loading state', () => {
  const loading = true;
  const tree = renderer.create(<ButtonNext loading={loading} onButtonClick={functionMock} />).toJSON();
  expect(tree).toMatchSnapshot();
});