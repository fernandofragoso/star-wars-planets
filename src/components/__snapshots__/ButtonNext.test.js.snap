// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<button
  className="button"
  onClick={[Function]}
>
  NEXT
</button>
`;

exports[`renders correctly on loading state 1`] = `
<button
  className="button--disabled"
  onClick={[Function]}
>
  NEXT
</button>
`;
